### PR TITLE
#162685676 User can share articles across social media platforms

### PR DESCRIPTION
**Pull request by fahadmak**
#### What does this PR do?
PR enables users should be able to share articles across all social media platforms.
#### Description of Task to be completed?
A logged in/authenticated user should be able to share articles across all social media platforms.
#### How should this be manually tested?
Before testing run these commands in cmd

`git clone https://github.com/andela/Ah-backend-guardians.git`

`cd Ah-backend-guardians`

`git checkout ft-article-read-time-162685675`

`virtualenv env`

`. env/bin/activate`

`pip install -r requirements.txt`

1.  Create a Postgres database and a `.env` file in your root directory and add the following configurations.
```
SECRET_KEY=andela_sjdndkjghdf78872

DB_NAME=your_database_name

DB_USER=database_user

DB_PASSWORD=database_password

DB_HOST=127.0.0.1

ALLOWED_HOSTS=.localhost, .herokuapp.com

EMAIL_PORT=587

EMAIL_HOST_USER = gauthorshaven@gmail.com

EMAIL_HOST_PASSWORD = guardiansauthors

WORD_LENGTH = 5

WPM = 200
```
2. Copy and paste this link into postman http://127.0.0.1:8000/api/articles/ to test the app using postman. First create the article by Sending POST request using this data below in body section
```
{
	
	"title": "An article",
	"body":"This is an article that does so many things",
	"description":"This is an article"
}
```

THen Go over to the drop-down and select the GET request to return all article's where you will the respective social links for a particular article
**This should be expected response** 
```{
    "article": {
        "id": 1,
        "title": "The man",
        "body": "Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmodtempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniamquis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodoconsequat. Duis aute irure dolor in reprehenderit in voluptate velit essecillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat nonproident, sunt in culpa qui officia deserunt mollit anim id est laborum.",
        "description": "it's crazy man",
        "author": {
            "email": "f.faraqhans2@gmail.com",
            "username": "fahads2"
        },
        "slug": "the-man",
        "published": true,
        "created_at": "2019-01-23T15:34:50.888407Z",
        "updated_at": "2019-01-23T15:34:50.888428Z",
        "images": "",
        "read_time": null,
        "tags": [],
        "social_links": {
            "facebook": "https://www.facebook.com/sharer/sharer.php?u=http%3A//127.0.0.1%3A8000/api/articles/the-man/",
            "twitter": "https://twitter.com/home?status=http%3A//127.0.0.1%3A8000/api/articles/the-man/",
            "email": "mailto:?&subject=The%20man%20from%20Author%27s%20Haven&body=Click%20Link%20To%20View%20The%20Article%20http%3A//127.0.0.1%3A8000/api/articles/the-man/"
        }
    }
}
```
#### What are the relevant pivotal tracker stories?
[#162685671](https://www.pivotaltracker.com/story/show/162685671)

### CODE REVIEW

**Review by myrdstom — APPROVED** [review]
LGTM

File: authors/apps/articles/serializers.py [hunk 1]
```diff
@@ -1,3 +1,6 @@
+from urllib.parse import quote_plus, quote
```
Comment (Opio-Emmanuel-Omona) [inline authors/apps/articles/serializers.py]: @fahadmak You imported quote_plus but never used it.

Comment (fahadmak) [reply authors/apps/articles/serializers.py]: Thanx for the observation let me work on it

**Review by tomuhenry — APPROVED** [review]
LGTM...

**Review by Opio-Emmanuel-Omona — APPROVED** [review]

**Comment by archbishop12** [comment]
@fahadmak Look into this. Twitter is fine. but facebook 

# Screenshot 

screen shot 2019-01-28 at 13 08 23

**Review by archbishop12 — APPROVED** [review]
@fahadmak LGTM I am merging this.

**Pull request merged**